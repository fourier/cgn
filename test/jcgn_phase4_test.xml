<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn" 
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java">

  <!-- this test tests the 4rd java transformation phase -->

  <!-- test if the the field jcgn:date-type is not set, but the
       cgn:type is 'date' or '[date]', take the jcgn:date-type
       from the parent cgn:object
  -->
  <objects cgn:package="com.test.java.phase4.field.datetype.notset">
    <object cgn:name="Test1" jcgn:date-type="joda">
      <field cgn:name="java-date" cgn:type="date" jcgn:date-type="java"/>
      <field cgn:name="java-array" cgn:type="[date]" jcgn:date-type="java"/>
      <field cgn:name="joda-date" cgn:type="date" />
      <field cgn:name="joda-array" cgn:type="[date]"/>
    </object>
  </objects>

</root>

