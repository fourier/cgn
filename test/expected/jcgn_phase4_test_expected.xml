<?xml version="1.0" encoding="UTF-8"?>
<!-- default namespace for elements: cgn -->
<root xmlns="https://github.com/fourier/cgn"
      xmlns:cgn="https://github.com/fourier/cgn"
      xmlns:jcgn="https://github.com/fourier/cgn/java"><!-- this test tests the 4rd java transformation phase --><!-- test if the the field jcgn:date-type is not set, but the
       cgn:type is 'date' or '[date]', take the jcgn:date-type
       from the parent cgn:object
  -->
   <objects cgn:package="com.test.java.phase4.field.datetype.notset"
            cgn:copyright=""
            cgn:author=""
            cgn:read-only="false"
            cgn:json="false"
            cgn:is-set="false"
            jcgn:builder="false"
            jcgn:parcelable="false"
            jcgn:date-type="java">
      <object cgn:name="Test1"
              jcgn:date-type="joda"
              cgn:copyright=""
              cgn:author=""
              cgn:package="com.test.java.phase4.field.datetype.notset"
              cgn:read-only="false"
              cgn:json="false"
              cgn:is-set="false"
              jcgn:builder="false"
              jcgn:parcelable="false">
         <field cgn:name="java-date" cgn:type="date" jcgn:date-type="java"/>
         <field cgn:name="java-array" cgn:type="[date]" jcgn:date-type="java"/>
         <field cgn:name="joda-date" cgn:type="date" jcgn:date-type="joda"/>
         <field cgn:name="joda-array" cgn:type="[date]" jcgn:date-type="joda"/>
      </object>
   </objects>
</root>
